### PR TITLE
Target problem

### PR DESCRIPTION
**Pull request by btry**

### CODE REVIEW

File: inc/targetproblem.class.php [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+         'content',
+         'impactcontent',
+         'causecontent',
+         'symptomcontent',
+      ];
+   }
+
+   public function prepareInputForUpdate($input) {
+      // Control fields values :
+      if (!isset($input['_skip_checks'])
```
Comment (AdrienClairembault) [inline inc/targetproblem.class.php:341]: Is it safe to handle `_skip_checks` directly in the input ?
Wouldn't it be possible for a malicious user to insert it into his form and save unchecked data ?

**Review by AdrienClairembault — APPROVED** [review]
Tested the feature, seems to work as expected 👍

File: inc/targetproblem.class.php [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+         // Not used for now
+         $title =  __('Add a target ', 'formcreator');
+      } else {
+         $title =  __('Edit a target', 'formcreator');
+      }
+      $rand = mt_rand();
+
+      $form = $this->getForm();
+
+      // TODO: remive the fixed width
```
[the remark this answers is not in the document]
Comment (btry) [reply inc/targetproblem.class.php:94]: This TODO is actually obsolete, the task was done while updating the UI for GLPI 10

see in the branch support/2.12.0 :

image

File: inc/targetproblem.class.php [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+      . ' action="' . self::getFormURL() . '"'
+      . ' data-itemtype="' . self::class . '"'
+      . '>';
+
+      // General information: target_name
+      echo '<table class="tab_cadre_fixe">';
+      echo '<tr><th colspan="2">' . $title . '</th></tr>';
+      echo '<tr>';
+      echo '<td width="15%"><strong>' . __('Name') . ' <span style="color:red;">*</span></strong></td>';
+      // TODO: remive the fixed width
```
[the remark this answers is not in the document]
Comment (btry) [reply inc/targetproblem.class.php:106]: same as above

[the remark this answers is not in the document]
Comment (btry) [reply inc/targetproblem.class.php:106]: here, the change was not done. I'll add some updates to fix that

**Review by AdrienClairembault — APPROVED** [review]

**Pull request merged**